subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java

```diff
@@ -157,4 +168,36 @@ public boolean isAllowInsecureProtocol() {
     public void setAllowInsecureProtocol(boolean allowInsecureProtocol) {
         this.allowInsecureProtocol = allowInsecureProtocol;
     }
+
+    /**
+     * Specifies whether HTTP expect-continue should be used for store requests.
+     *
+     * This value defaults to {@code false}.
+     *
+     * When enabled, whether or not a store request would succeed is checked with the server before attempting.
+     * This is particularly useful when potentially dealing with large artifacts that may be rejected by the server,
+     * as it avoids the overhead of transmitting the large file just to have it rejected.
+     * This fail-fast behavior comes at the expense of extra marginal overhead for successful requests,
+     * due to the extra network communication required by the initial check.
+     *
+     * Note: not all HTTP servers support expect-continue.
+     *
+     * @param useExpectContinue whether HTTP expect-continue should be used for store requests
+     * @since 7.2
+     */
+    @Incubating
+    public void setUseExpectContinue(boolean useExpectContinue) {
+        this.useExpectContinue = useExpectContinue;
+    }
+
+    /**
+     * Whether HTTP expect-continue should be used for store requests.
+     *
+     * @see #setUseExpectContinue(boolean)
+     * @since 7.2
+     */
+    @Incubating
+    public boolean isUseExpectContinue() {
+        return useExpectContinue;
+    }
```

Comment:
For `allowInsecureProtocol` the getter's docs have the extra details, while the setter's link to the getter. I think it would be nice to keep this consistent.

(We should also move to using `Property<Boolean>` later, which would allow documenting the feature in a single location, but that's a different story.)

Reply:
Changed this and the other existing properties to use this pattern.

Is this the general pattern, or just for this class?

Reply:
Don't know, I just noticed that we have it here. I think what we should be doing is moving to `Property` instances instead, which will remove this problem entirely.